### PR TITLE
View-refactor: Move definition of ALL

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

**Review by dalg24 — CHANGES_REQUESTED** [review]
You forgot to remove https://github.com/kokkos/kokkos/blob/9e875704e09697713ba51f9e5a990166fb679b97/core/src/View/Kokkos_ViewMapping.hpp#L44-L55

File: core/src/View/Kokkos_ViewTraits.hpp [hunk 1]
```diff
@@ -32,6 +32,37 @@ static_assert(false,
 //----------------------------------------------------------------------------
 
 namespace Kokkos {
+
+struct ALL_t {
+  KOKKOS_INLINE_FUNCTION
```
Comment (dalg24) [inline core/src/View/Kokkos_ViewTraits.hpp:37]: How about we change KOKKOS_INLINE_FUNCTION -> KOKKOS_FUNCTION while we are at it?

File: core/src/View/Kokkos_ViewTraits.hpp [hunk 2]
```diff
@@ -32,6 +32,37 @@ static_assert(false,
...
+
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+namespace Impl {
+// TODO This alias declaration forces us to fully qualify ALL_t inside the
+// Kokkos::Impl namespace to avoid deprecation warnings. Replace the
+// fully-qualified name when we remove Kokkos::Impl::ALL_t.
+using ALL_t KOKKOS_DEPRECATED_WITH_COMMENT("Use Kokkos::ALL_t instead!") =
+    Kokkos::ALL_t;
+}  // namespace Impl
+#endif
```
Comment (dalg24) [inline core/src/View/Kokkos_ViewTraits.hpp:52]:
```suggestion
#endif

```

**Review by dalg24 — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]

**Comment by masterleinad** [comment]
You'll have to fix the formatting:
```diff
diff --git a/core/src/View/Kokkos_ViewTraits.hpp b/core/src/View/Kokkos_ViewTraits.hpp
index 48995f6bd..d04d347c8 100644
--- a/core/src/View/Kokkos_ViewTraits.hpp
+++ b/core/src/View/Kokkos_ViewTraits.hpp
@@ -63,7 +63,6 @@ inline constexpr Kokkos::ALL_t ALL{};
 #pragma omp end declare target
 #endif
 
-
 namespace Impl {
 
 template <class DataType>
```

File: core/src/View/Kokkos_ViewTraits.hpp [hunk 3]
```diff
@@ -32,6 +32,38 @@ static_assert(false,
...
+#pragma omp declare target
+#endif
+
+inline constexpr Kokkos::ALL_t ALL{};
+
+#if defined(KOKKOS_ENABLE_OPENMPTARGET) && defined(KOKKOS_COMPILER_INTEL_LLVM)
+#pragma omp end declare target
+#endif
+
+
```
Comment (dalg24) [inline core/src/View/Kokkos_ViewTraits.hpp:66]:
```suggestion
```

**Comment by dalg24** [comment]
Unrelated failures (one CUDA build not seeing GPU at runtime and nvhpc as host segfaulting)

**Pull request merged**